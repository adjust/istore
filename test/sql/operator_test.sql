BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>0'::bigistore -> -1;
SELECT '1=>1, -1=>3'::bigistore -> -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore ? -1;
SELECT '1=>1, -1=>3'::bigistore ? 5;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore + '1=>1'::bigistore;
SELECT '1=>1, -1=>3'::bigistore + '-1=>-1'::bigistore;
SELECT '1=>1, -1=>3'::bigistore + '1=>-1'::bigistore;
SELECT '1=>0, -1=>3'::bigistore + '1=>-1'::bigistore;
SELECT '1=>1, -1=>0'::bigistore + '-1=>-1'::bigistore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '0=>40000000000'::bigistore->0;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore  + 1;
SELECT '-1=>1, 1=>3'::bigistore  + 1;
SELECT '-1=>1, -1=>3'::bigistore + 1;
SELECT '1=>1, -1=>3'::bigistore  + 0;
SELECT '-1=>1, 1=>3'::bigistore  + 0;
SELECT '-1=>1, -1=>3'::bigistore + 0;
SELECT '1=>1, -1=>3'::bigistore  + -1;
SELECT '-1=>1, 1=>3'::bigistore  + -1;
SELECT '-1=>1, -1=>3'::bigistore + -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore - '1=>1'::bigistore;
SELECT '1=>1, -1=>3'::bigistore - '-1=>-1'::bigistore;
SELECT '1=>1, -1=>3'::bigistore - '1=>-1'::bigistore;
SELECT '1=>0, -1=>3'::bigistore - '1=>-1'::bigistore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore  - 1;
SELECT '-1=>1, 1=>3'::bigistore  - 1;
SELECT '-1=>1, -1=>3'::bigistore - 1;
SELECT '1=>1, -1=>3'::bigistore  - 0;
SELECT '-1=>1, 1=>3'::bigistore  - 0;
SELECT '-1=>1, -1=>3'::bigistore - 0;
SELECT '1=>1, -1=>3'::bigistore  - -1;
SELECT '-1=>1, 1=>3'::bigistore  - -1;
SELECT '-1=>1, -1=>3'::bigistore - -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>3, 2=>2'::bigistore * '1=>2, 3=>5'::bigistore;
SELECT '-1=>3, 2=>2'::bigistore * '-1=>2, 3=>5'::bigistore;
SELECT '-1=>3, 2=>2'::bigistore * '-1=>-2, 3=>5'::bigistore;
SELECT '-1=>3, 2=>0'::bigistore * '-1=>-2, 3=>5'::bigistore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore  * 1;
SELECT '-1=>1, 1=>3'::bigistore  * 1;
SELECT '-1=>1, -1=>3'::bigistore * 1;
SELECT '1=>1, -1=>3'::bigistore  * 0;
SELECT '-1=>1, 1=>3'::bigistore  * 0;
SELECT '-1=>1, -1=>3'::bigistore * 0;
SELECT '1=>1, -1=>3'::bigistore  * -1;
SELECT '-1=>1, 1=>3'::bigistore  * -1;
SELECT '-1=>1, -1=>3'::bigistore * -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>0'::istore -> -1;
SELECT '1=>1, -1=>3'::istore -> -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore ? -1;
SELECT '1=>1, -1=>3'::istore ? 5;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore + '1=>1'::istore;
SELECT '1=>1, -1=>3'::istore + '-1=>-1'::istore;
SELECT '1=>1, -1=>3'::istore + '1=>-1'::istore;
SELECT '1=>0, -1=>3'::istore + '1=>-1'::istore;
SELECT '1=>1, -1=>0'::istore + '-1=>-1'::istore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore  + 1;
SELECT '-1=>1, 1=>3'::istore  + 1;
SELECT '-1=>1, -1=>3'::istore + 1;
SELECT '1=>1, -1=>3'::istore  + 0;
SELECT '-1=>1, 1=>3'::istore  + 0;
SELECT '-1=>1, -1=>3'::istore + 0;
SELECT '1=>1, -1=>3'::istore  + -1;
SELECT '-1=>1, 1=>3'::istore  + -1;
SELECT '-1=>1, -1=>3'::istore + -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore - '1=>1'::istore;
SELECT '1=>1, -1=>3'::istore - '-1=>-1'::istore;
SELECT '1=>1, -1=>3'::istore - '1=>-1'::istore;
SELECT '1=>0, -1=>3'::istore - '1=>-1'::istore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore  - 1;
SELECT '-1=>1, 1=>3'::istore  - 1;
SELECT '-1=>1, -1=>3'::istore - 1;
SELECT '1=>1, -1=>3'::istore  - 0;
SELECT '-1=>1, 1=>3'::istore  - 0;
SELECT '-1=>1, -1=>3'::istore - 0;
SELECT '1=>1, -1=>3'::istore  - -1;
SELECT '-1=>1, 1=>3'::istore  - -1;
SELECT '-1=>1, -1=>3'::istore - -1;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>3, 2=>2'::istore * '1=>2, 3=>5'::istore;
SELECT '-1=>3, 2=>2'::istore * '-1=>2, 3=>5'::istore;
SELECT '-1=>3, 2=>2'::istore * '-1=>-2, 3=>5'::istore;
SELECT '-1=>3, 2=>0'::istore * '-1=>-2, 3=>5'::istore;
ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore  * 1;
SELECT '-1=>1, 1=>3'::istore  * 1;
SELECT '-1=>1, -1=>3'::istore * 1;
SELECT '1=>1, -1=>3'::istore  * 0;
SELECT '-1=>1, 1=>3'::istore  * 0;
SELECT '-1=>1, -1=>3'::istore * 0;
SELECT '1=>1, -1=>3'::istore  * -1;
SELECT '-1=>1, 1=>3'::istore  * -1;
SELECT '-1=>1, -1=>3'::istore * -1;
ROLLBACK;

-- ISTORE = and <> operators
BEGIN;
CREATE EXTENSION istore;
-- = TRUE
SELECT '1=>1, -1=>0'::istore = '1=>1, -1=>0'::istore;
SELECT '1=>1, -1=>0'::istore = '-1=>0, 1=>1'::istore;
SELECT '1=>1, 1=>1'::istore  = '1=>2'::istore;
SELECT ''::istore = ''::istore;
SELECT NULLIF('1=>1, -1=>100'::istore, '1=>1, -1=>0'::istore);
-- <> TRUE
SELECT '1=>1, -1=>0'::istore <> '1=>1'::istore;
-- = FALSE
SELECT '1=>1, -1=>0, -2=>-2'::istore = '1=>1, -1=>0'::istore;
-- <> FALSE
SELECT '1=>1, -1=>0'::istore != '1=>1, -1=>0'::istore;
-- NULL cases
SELECT (NULL::istore = '1=>1, -1=>0'::istore) is null;
SELECT (NULL::istore = NULL::istore) is null;
SELECT (NULLIF('1=>1, -1=>0'::istore, '1=>1, -1=>0'::istore)) is null;
ROLLBACK;

-- BIGISTORE = and <> operators
BEGIN;
CREATE EXTENSION istore;
-- = TRUE
SELECT '1=>1234123412341234, -1=>1234123412341234'::bigistore = '1=>1234123412341234, -1=>1234123412341234'::bigistore;
SELECT '1=>1, -1=>0'::bigistore = '-1=>0, 1=>1'::bigistore;
SELECT '1=>1, 1=>1'::bigistore  = '1=>2'::bigistore;
SELECT ''::bigistore = ''::bigistore;
SELECT NULLIF('1=>1, -1=>100'::bigistore, '1=>1, -1=>0'::bigistore);
-- <> TRUE
SELECT '1=>1, -1=>0'::bigistore <> '1=>1'::bigistore;
-- = FALSE
SELECT '1=>1234123412341234, -1=>1234123412341234'::bigistore = '100=>1234123412341234, -100=>1234123412341234'::bigistore;
-- <> FALSE
SELECT '1=>1, -1=>0'::bigistore != '1=>1, -1=>0'::bigistore;
-- NULL cases
SELECT (NULL::bigistore = '1=>1, -1=>0'::bigistore) is null;
SELECT (NULL::bigistore = NULL::bigistore) is null;
SELECT (NULLIF('1=>1, -1=>0'::bigistore, '1=>1, -1=>0'::bigistore)) is null;
ROLLBACK;
