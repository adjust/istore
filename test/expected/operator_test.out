BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>0'::bigistore -> -1;
 ?column? 
----------
        0
(1 row)

SELECT '1=>1, -1=>3'::bigistore -> -1;
 ?column? 
----------
        3
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore ? -1;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, -1=>3'::bigistore ? 5;
 ?column? 
----------
 f
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore + '1=>1'::bigistore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::bigistore + '-1=>-1'::bigistore;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::bigistore + '1=>-1'::bigistore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"0"
(1 row)

SELECT '1=>0, -1=>3'::bigistore + '1=>-1'::bigistore;
       ?column?       
----------------------
 "-1"=>"3", "1"=>"-1"
(1 row)

SELECT '1=>1, -1=>0'::bigistore + '-1=>-1'::bigistore;
       ?column?       
----------------------
 "-1"=>"-1", "1"=>"1"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '0=>40000000000'::bigistore->0;
  ?column?   
-------------
 40000000000
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore  + 1;
      ?column?       
---------------------
 "-1"=>"4", "1"=>"2"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  + 1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"4"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore + 1;
 ?column?  
-----------
 "-1"=>"5"
(1 row)

SELECT '1=>1, -1=>3'::bigistore  + 0;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  + 0;
      ?column?       
---------------------
 "-1"=>"1", "1"=>"3"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore + 0;
 ?column?  
-----------
 "-1"=>"4"
(1 row)

SELECT '1=>1, -1=>3'::bigistore  + -1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"0"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  + -1;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"2"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore + -1;
 ?column?  
-----------
 "-1"=>"3"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore - '1=>1'::bigistore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::bigistore - '-1=>-1'::bigistore;
      ?column?       
---------------------
 "-1"=>"4", "1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::bigistore - '1=>-1'::bigistore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"2"
(1 row)

SELECT '1=>0, -1=>3'::bigistore - '1=>-1'::bigistore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore  - 1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"0"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  - 1;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"2"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore - 1;
 ?column?  
-----------
 "-1"=>"3"
(1 row)

SELECT '1=>1, -1=>3'::bigistore  - 0;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  - 0;
      ?column?       
---------------------
 "-1"=>"1", "1"=>"3"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore - 0;
 ?column?  
-----------
 "-1"=>"4"
(1 row)

SELECT '1=>1, -1=>3'::bigistore  - -1;
      ?column?       
---------------------
 "-1"=>"4", "1"=>"2"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  - -1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"4"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore - -1;
 ?column?  
-----------
 "-1"=>"5"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>3, 2=>2'::bigistore * '1=>2, 3=>5'::bigistore;
 ?column? 
----------
 "1"=>"6"
(1 row)

SELECT '-1=>3, 2=>2'::bigistore * '-1=>2, 3=>5'::bigistore;
 ?column?  
-----------
 "-1"=>"6"
(1 row)

SELECT '-1=>3, 2=>2'::bigistore * '-1=>-2, 3=>5'::bigistore;
  ?column?  
------------
 "-1"=>"-6"
(1 row)

SELECT '-1=>3, 2=>0'::bigistore * '-1=>-2, 3=>5'::bigistore;
  ?column?  
------------
 "-1"=>"-6"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::bigistore  * 1;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  * 1;
      ?column?       
---------------------
 "-1"=>"1", "1"=>"3"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore * 1;
 ?column?  
-----------
 "-1"=>"4"
(1 row)

SELECT '1=>1, -1=>3'::bigistore  * 0;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"0"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  * 0;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"0"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore * 0;
 ?column?  
-----------
 "-1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::bigistore  * -1;
       ?column?        
-----------------------
 "-1"=>"-3", "1"=>"-1"
(1 row)

SELECT '-1=>1, 1=>3'::bigistore  * -1;
       ?column?        
-----------------------
 "-1"=>"-1", "1"=>"-3"
(1 row)

SELECT '-1=>1, -1=>3'::bigistore * -1;
  ?column?  
------------
 "-1"=>"-4"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>0'::istore -> -1;
 ?column? 
----------
        0
(1 row)

SELECT '1=>1, -1=>3'::istore -> -1;
 ?column? 
----------
        3
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore ? -1;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, -1=>3'::istore ? 5;
 ?column? 
----------
 f
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore + '1=>1'::istore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"2"
(1 row)

SELECT '1=>1, -1=>3'::istore + '-1=>-1'::istore;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore + '1=>-1'::istore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"0"
(1 row)

SELECT '1=>0, -1=>3'::istore + '1=>-1'::istore;
       ?column?       
----------------------
 "-1"=>"3", "1"=>"-1"
(1 row)

SELECT '1=>1, -1=>0'::istore + '-1=>-1'::istore;
       ?column?       
----------------------
 "-1"=>"-1", "1"=>"1"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore  + 1;
      ?column?       
---------------------
 "-1"=>"4", "1"=>"2"
(1 row)

SELECT '-1=>1, 1=>3'::istore  + 1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"4"
(1 row)

SELECT '-1=>1, -1=>3'::istore + 1;
 ?column?  
-----------
 "-1"=>"5"
(1 row)

SELECT '1=>1, -1=>3'::istore  + 0;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

SELECT '-1=>1, 1=>3'::istore  + 0;
      ?column?       
---------------------
 "-1"=>"1", "1"=>"3"
(1 row)

SELECT '-1=>1, -1=>3'::istore + 0;
 ?column?  
-----------
 "-1"=>"4"
(1 row)

SELECT '1=>1, -1=>3'::istore  + -1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"0"
(1 row)

SELECT '-1=>1, 1=>3'::istore  + -1;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"2"
(1 row)

SELECT '-1=>1, -1=>3'::istore + -1;
 ?column?  
-----------
 "-1"=>"3"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore - '1=>1'::istore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore - '-1=>-1'::istore;
      ?column?       
---------------------
 "-1"=>"4", "1"=>"1"
(1 row)

SELECT '1=>1, -1=>3'::istore - '1=>-1'::istore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"2"
(1 row)

SELECT '1=>0, -1=>3'::istore - '1=>-1'::istore;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore  - 1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"0"
(1 row)

SELECT '-1=>1, 1=>3'::istore  - 1;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"2"
(1 row)

SELECT '-1=>1, -1=>3'::istore - 1;
 ?column?  
-----------
 "-1"=>"3"
(1 row)

SELECT '1=>1, -1=>3'::istore  - 0;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

SELECT '-1=>1, 1=>3'::istore  - 0;
      ?column?       
---------------------
 "-1"=>"1", "1"=>"3"
(1 row)

SELECT '-1=>1, -1=>3'::istore - 0;
 ?column?  
-----------
 "-1"=>"4"
(1 row)

SELECT '1=>1, -1=>3'::istore  - -1;
      ?column?       
---------------------
 "-1"=>"4", "1"=>"2"
(1 row)

SELECT '-1=>1, 1=>3'::istore  - -1;
      ?column?       
---------------------
 "-1"=>"2", "1"=>"4"
(1 row)

SELECT '-1=>1, -1=>3'::istore - -1;
 ?column?  
-----------
 "-1"=>"5"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>3, 2=>2'::istore * '1=>2, 3=>5'::istore;
 ?column? 
----------
 "1"=>"6"
(1 row)

SELECT '-1=>3, 2=>2'::istore * '-1=>2, 3=>5'::istore;
 ?column?  
-----------
 "-1"=>"6"
(1 row)

SELECT '-1=>3, 2=>2'::istore * '-1=>-2, 3=>5'::istore;
  ?column?  
------------
 "-1"=>"-6"
(1 row)

SELECT '-1=>3, 2=>0'::istore * '-1=>-2, 3=>5'::istore;
  ?column?  
------------
 "-1"=>"-6"
(1 row)

ROLLBACK;
BEGIN;
CREATE EXTENSION istore;
SELECT '1=>1, -1=>3'::istore  * 1;
      ?column?       
---------------------
 "-1"=>"3", "1"=>"1"
(1 row)

SELECT '-1=>1, 1=>3'::istore  * 1;
      ?column?       
---------------------
 "-1"=>"1", "1"=>"3"
(1 row)

SELECT '-1=>1, -1=>3'::istore * 1;
 ?column?  
-----------
 "-1"=>"4"
(1 row)

SELECT '1=>1, -1=>3'::istore  * 0;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"0"
(1 row)

SELECT '-1=>1, 1=>3'::istore  * 0;
      ?column?       
---------------------
 "-1"=>"0", "1"=>"0"
(1 row)

SELECT '-1=>1, -1=>3'::istore * 0;
 ?column?  
-----------
 "-1"=>"0"
(1 row)

SELECT '1=>1, -1=>3'::istore  * -1;
       ?column?        
-----------------------
 "-1"=>"-3", "1"=>"-1"
(1 row)

SELECT '-1=>1, 1=>3'::istore  * -1;
       ?column?        
-----------------------
 "-1"=>"-1", "1"=>"-3"
(1 row)

SELECT '-1=>1, -1=>3'::istore * -1;
  ?column?  
------------
 "-1"=>"-4"
(1 row)

ROLLBACK;
-- ISTORE = and <> operators
BEGIN;
CREATE EXTENSION istore;
-- = TRUE
SELECT '1=>1, -1=>0'::istore = '1=>1, -1=>0'::istore;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, -1=>0'::istore = '-1=>0, 1=>1'::istore;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, 1=>1'::istore  = '1=>2'::istore;
 ?column? 
----------
 t
(1 row)

SELECT ''::istore = ''::istore;
 ?column? 
----------
 t
(1 row)

SELECT NULLIF('1=>1, -1=>100'::istore, '1=>1, -1=>0'::istore);
        nullif         
-----------------------
 "-1"=>"100", "1"=>"1"
(1 row)

-- <> TRUE
SELECT '1=>1, -1=>0'::istore <> '1=>1'::istore;
 ?column? 
----------
 t
(1 row)

-- = FALSE
SELECT '1=>1, -1=>0, -2=>-2'::istore = '1=>1, -1=>0'::istore;
 ?column? 
----------
 f
(1 row)

-- <> FALSE
SELECT '1=>1, -1=>0'::istore != '1=>1, -1=>0'::istore;
 ?column? 
----------
 f
(1 row)

-- NULL cases
SELECT (NULL::istore = '1=>1, -1=>0'::istore) is null;
 ?column? 
----------
 t
(1 row)

SELECT (NULL::istore = NULL::istore) is null;
 ?column? 
----------
 t
(1 row)

SELECT (NULLIF('1=>1, -1=>0'::istore, '1=>1, -1=>0'::istore)) is null;
 ?column? 
----------
 t
(1 row)

ROLLBACK;
-- BIGISTORE = and <> operators
BEGIN;
CREATE EXTENSION istore;
-- = TRUE
SELECT '1=>1234123412341234, -1=>1234123412341234'::bigistore = '1=>1234123412341234, -1=>1234123412341234'::bigistore;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, -1=>0'::bigistore = '-1=>0, 1=>1'::bigistore;
 ?column? 
----------
 t
(1 row)

SELECT '1=>1, 1=>1'::bigistore  = '1=>2'::bigistore;
 ?column? 
----------
 t
(1 row)

SELECT ''::bigistore = ''::bigistore;
 ?column? 
----------
 t
(1 row)

SELECT NULLIF('1=>1, -1=>100'::bigistore, '1=>1, -1=>0'::bigistore);
        nullif         
-----------------------
 "-1"=>"100", "1"=>"1"
(1 row)

-- <> TRUE
SELECT '1=>1, -1=>0'::bigistore <> '1=>1'::bigistore;
 ?column? 
----------
 t
(1 row)

-- = FALSE
SELECT '1=>1234123412341234, -1=>1234123412341234'::bigistore = '100=>1234123412341234, -100=>1234123412341234'::bigistore;
 ?column? 
----------
 f
(1 row)

-- <> FALSE
SELECT '1=>1, -1=>0'::bigistore != '1=>1, -1=>0'::bigistore;
 ?column? 
----------
 f
(1 row)

-- NULL cases
SELECT (NULL::bigistore = '1=>1, -1=>0'::bigistore) is null;
 ?column? 
----------
 t
(1 row)

SELECT (NULL::bigistore = NULL::bigistore) is null;
 ?column? 
----------
 t
(1 row)

SELECT (NULLIF('1=>1, -1=>0'::bigistore, '1=>1, -1=>0'::bigistore)) is null;
 ?column? 
----------
 t
(1 row)

ROLLBACK;
